packages/models/src/appMapBuilder/eventStack.js

```diff
@@ -46,7 +40,10 @@ export default class EventStack {
           call.parent = parent;
         }
       } else {
-        throw new Error(`return #${event.id} is missing call #${event.parent_id}`);
+        // This can happen in async contexts where the recording began in the middle
+        // of another async task. It's possible that the parent call was executed
+        // before the recording began. If this is the case, discard the return event.
+        return;
```

Comment:
It would be good idea to raise a warning, but I'm not sure where it should go...

Reply:
I thought the same thing. The issue is that this is called in so many different contexts it felt wrong throwing a `console.log` in there, and we don't have any other logging abstraction.

Reply:
Yeah, good discussion but the main thing is to just not crash :-)